### ISSUE TITLE
swagger-php 2.x support

### ISSUE THREAD

**Issue by JeroenDeDauw** [jdesrosiers/silex-swagger-provider#9]
Any plans to support swagger-php 2.x (and thus also Swagger 2.x)?

**Comment by jdesrosiers** [comment]
To be honest, I wasn't planning on it.  I'm not really a fan of Swagger anymore, so this project doesn't get a lot of attention.  If there is demand for it, I might be able to get to it some time in December.  I also accept pull requests.

**Comment by JeroenDeDauw** [comment]
Thanks for the quick reply

> I'm not really a fan of Swagger anymore

Why not, and is there something else you can recommend instead?

**Comment by jdesrosiers** [comment]
Are you familiar with the Richardson Maturity Model (http://martinfowler.com/articles/richardsonMaturityModel.html)?  Swagger is designed to support Level 2 REST APIs.  REST is supposed to support decoupling client and server.  By defining inputs and outputs for operations the way Swagger does, it couples the client and server.  In REST, a resource itself should describe what operations can be performed on it.

I use JSON Hyper-Schema as an alternative to Swagger and Jsonary as an alternative to Swagger-ui.  It's not nearly as pretty, but it allows me to write Level 3 REST APIs.

**Comment by JeroenDeDauw** [comment]
Thanks for explaining, and mentioning those alternatives.

I'm in the early stages of creating the API I'm working on, though it already uses hypermedia controls and will continue to do so.

I'm new to Swagger and have not actually written my spec yet, so this might be a dumb question: is it the case that Swagger does not provide support for documenting hypermedia controls, or does it somehow prevent you from having them?

**Comment by jdesrosiers** [comment]
Swagger can't prevent you from doing anything.  It can only describe your API.  It's just that it isn't designed to describe hypermedia driven APIs, so it's tools tend to fall short.

**Comment by tarlepp** [comment]
Darn, I was really looking for update to use newer version of swagger with this. @jdesrosiers Is there any Silex providers to document level 3 REST APIs?

**Comment by jdesrosiers** [comment]
I use JSON Hyper-Schema for Hypermedia API work.  I've compiled some basic JSON Schema tools in [jdesrosiers/silex-json-schema-provider](https://github.com/jdesrosiers/silex-json-schema-provider).  I don't know if that is the kind of thing you are looking for, but it's what I use.

**Comment by joaocc** [comment]
Hi @jdesrosiers. 
I noticed you have commits to silex-2 branch... 
Did you reconsider your luv for the users of 2.0? :)
Seriously, does this mean you will be updating this for 2.0?
Thx

**Comment by jdesrosiers** [comment]
No, the silex-2 branch was for supporting Silex 2.0, not Swagger 2.0, but it doesn't matter anyway.  This project is effectively dead.  I'll take pull requests or even transfer ownership if someone wants to take over, but I don't have time or interest to maintain something I no longer use.

**Comment by tarlepp** [comment]
darn
